activedata_etl/transforms/jscov_to_es.py

```diff
@@ -13,61 +13,120 @@
+                counter = count_generator().next
+                for source_file_index, obj in enumerate(stream.parse(json_stream, '.', ['.'])):
+                    if please_stop:
+                        Log.error("Shutdown detected. Stopping job ETL.")
+
+                    if source_file_index == 0:
+                        # this is not a jscov object but an object containing the version metadata
+                        # TODO: this metadata should not be here
+                        # TODO: this version info is not used right now. Make use of it later.
+                        jscov_format_version = obj.get("version")
+                        continue
+
+                    try:
+                        for d in process_source_file(
+                            artifact_etl,
+                            counter,
+                            obj,
+                            task_cluster_record
+                        ):
+                            yield value2json(d)
+                    except Exception as e:
+                        Log.warning(
+                            "Error processing test {{test_url}} and source file {{source}} while processing {{key}}",
+                            key=source_key,
+                            test_url=obj.get("testUrl"),
+                            source=obj.get("sourceFile"),
+                            cause=e
+                        )
+
+    def aggregator():
+        aggr_coverage = dict()
+
+        with ZipFile(jsdcov_file) as zipped:
+            for num, zip_name in enumerate(zipped.namelist()):
+                json_stream = ibytes2ilines(zipped.open(zip_name))
+                for source_file_index, obj in enumerate(stream.parse(json_stream, '.', ['.'])):
+                    if please_stop:
+                        Log.error("Shutdown detected. Stopping job ETL.")
+
+                    if source_file_index == 0:
+                        # this is not a jscov object but an object containing the version metadata
+                        # TODO: this metadata should not be here
+                        # TODO: this version info is not used right now. Make use of it later.
+                        jscov_format_version = obj.get("version")
+                        continue
+
+                    obj = wrap(obj)
+
+                    # Collecting coverage information
+                    if obj.sourceFile in aggr_coverage:
+                        covered, uncovered = aggr_coverage[obj.sourceFile]
+                        covered.update(covered.union(set(obj.covered)))
+                        uncovered.update(uncovered.union(set(obj.uncovered)))
+                        aggr_coverage[obj.sourceFile] = (covered, uncovered)
+                    else:
+                        aggr_coverage[obj.sourceFile] = (set(obj.covered), set(obj.uncovered))
+
+        # Generate coverage information per source file
+        for source_file in aggr_coverage:
```

Comment:
instead of 

    for source_file in aggr_coverage:
        covered, uncovered = aggr_coverage[source_file]
        uncovered.update(uncovered - covered)

you can iterate over the key/value pairs with `items()`

    for source_file, agg in aggr_coverage.items():
        covered, uncovered = agg
        uncovered.update(uncovered - covered)

even better; you can pattern match `agg` too:

    for source_file, (covered, uncovered) in aggr_coverage.items():
        uncovered.update(uncovered - covered)

well, assuming that last line is correct

Reply:
Good tip! Thanks. I will make this change.